### PR TITLE
Helm: Add E2E for helm charts

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR adds the support for helm installation and to test cephcsi helm charts in Travis CI

fixes: https://github.com/ceph/ceph-csi/issues/526

### CODE REVIEW

File: scripts/install-helm.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#!/bin/bash -e
+
+#Based on ideas from https://github.com/rook/rook/blob/master/tests/scripts/helm.sh
+
+temp="/tmp/ceph-csi-tests-scripts-helm"
```
Comment (nixpanic) [inline scripts/install-helm.sh:5]: please use a real tmp dir if you can (not sure how to do the cleanup in that case)
```
temp=$(mktemp -d)
```

Also try not to mix lowercase and UPPERCASE variables.

File: scripts/travis-helmtest.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+sudo scripts/minikube.sh up
+sudo scripts/minikube.sh deploy-rook
+# pull docker images to speed up e2e
+sudo scripts/minikube.sh cephcsi
+sudo scripts/minikube.sh k8s-sidecar
+sudo chown -R travis: "$HOME"/.minikube /usr/local/bin/kubectl
+
+NAMESPACE=cephcsi-e2e-$RANDOM
+# set up helm
+sudo script/install-helm.sh up
```
Comment (nixpanic) [inline scripts/travis-helmtest.sh:16]: the directory is called `scripts/` (with an 's')

File: scripts/install-helm.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+install() {
+    if ! helm_loc="$(type -p "helm")" || [[ -z ${helm_loc} ]]; then
+        # Download and unpack helm
+        local dist
+        dist="$(uname -s)"
+        # shellcheck disable=SC2021
+        dist=$(echo "${dist}" | tr "[A-Z]" "[a-z]")
+        mkdir -p "${temp}"
```
Comment (nixpanic) [inline scripts/install-helm.sh:42]: this should be created by `mktemp -d`

File: scripts/install-helm.sh [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    if [ "${helm_ready}" != "Running" ]; then
+        echo "Helm init not successful"
+        exit 1
+    fi
+
+    echo "Helm init successful"
+
+}
+
+install-cephcsi-helm-charts() {
```
Comment (nixpanic) [inline scripts/install-helm.sh:80]: function names should not include `-`, use `_` instead.

File: scripts/install-helm.sh [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    if [ -z "$NAMESPACE" ]; then
+        NAMESPACE="default"
+    fi
+    # install ceph-csi-cephfs and ceph-csi-rbd charts
+    "${HELM}" install "${SCRIPT_DIR}"/../charts/ceph-csi-cephfs --name ${CEPHFS_CHART_NAME} --namespace ${NAMESPACE}
+    "${HELM}" install "${SCRIPT_DIR}"/../charts/ceph-csi-rbd --name ${RBD_CHART_NAME} --namespace ${NAMESPACE}
+}
+
+cleanup-cephcsi-charts() {
+    helm del --purge ${CEPHFS_CHART_NAME}
```
Comment (nixpanic) [inline scripts/install-helm.sh:91]: use `${HELM}`

File: scripts/travis-helmtest.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+# install cephcsi helm charts
+sudo script/install-helm.sh install-cephcsi ${NAMESPACE}
```
Comment (nixpanic) [inline scripts/travis-helmtest.sh:18]: does this really need `sudo`?

File: scripts/install-helm.sh [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        # shellcheck disable=SC2021
+        dist=$(echo "${dist}" | tr "[A-Z]" "[a-z]")
+        mkdir -p "${temp}"
+        wget "https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}-${dist}-${arch}.tar.gz" -O "${temp}/helm.tar.gz"
+        tar -C "${temp}" -zxvf "${temp}/helm.tar.gz"
+        HELM="${temp}/${dist}-${arch}/helm"
+    fi
+
+    # set up RBAC for helm
+    kubectl --namespace kube-system create sa tiller
```
Comment (nixpanic) [inline scripts/install-helm.sh:49]: Just wondering, why `tiller`?

Comment (Madhu-1) [reply scripts/install-helm.sh:48]: we need a tiller to install charts. https://v2.helm.sh/docs/install/#installing-tiller

File: .travis.yml [hunk 8]
```diff
@@ -8,6 +8,7 @@ addons:
     packages:
       - realpath
       - ruby
+      - socat
```
Comment (nixpanic) [inline .travis.yml:11]: Where is `socat` used? I can not see the need for it from the commit message.

Comment (Madhu-1) [reply .travis.yml:11]: This is needed for the helm chart. as this was the one for the requirement for helm chart and kubernetes didn't mention that in the commit message

Comment (Madhu-1) [reply .travis.yml:11]: @nixpanic do you want it as a separate commit?

Comment (Madhu-1) [reply .travis.yml:11]: updated the commit message for socat

**Pull request merged**